README.md

```diff
@@ -1,30 +1,127 @@
 # WIP: Cerializer
-Cerializer aims at providing a even faster alternative to FastAvro and Avro standard library.
+Cerializer is an Avro de/serialization library that aims at providing a even faster alternative to FastAvro and Avro standard library.
 
-This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
+This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made Cython code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
 
 Special credit needs to be given to [FastAvro](https://github.com/fastavro/fastavro) library, by which is this project heavily inspired.
 
+**Example of a schema and the corresponding code**
+
+SCHEMA
+```
+{
+    'name': 'array_schema', 
+    'doc': 'Array schema', 
+    'namespace': 'messaging', 
+    'type': 'record', 
+    'fields': [
+        {
+            'name': 'order_id', 
+            'doc': 'Id of order', 
+            'type': 'string'
+        }, 
+        {
+            'name': 'trades', 
+            'type': {
+                'type': 'array', 
+                'items': ['string', 'int']
+            }
+        }
+    ]
+}
+```
+
+CORRESPONDING CODE
+```
+def serialize(data, output):
+    cdef bytearray buffer = bytearray()
+    cdef dict datum
+    cdef str type_51
+
+    write.write_string(buffer, data['order_id'])
+
+    if len(data['trades']) > 0:
+        write.write_long(buffer, len(data['trades']))
+        for val_59 in data['trades']:
+            if type(val_59) is tuple:
+                type_51, val_60 = val_59
+                
+                if type_51 == 'string':
+                    write.write_long(buffer, 0)
+                    write.write_string(buffer, val_60)
+                
+                elif type_51 == 'int':
+                    write.write_long(buffer, 1)
+                    write.write_int(buffer, val_60)
+                
+
+            else:
+
+                if type(val_59) is str:
+                    write.write_long(buffer, 0)
+                    write.write_string(buffer, val_59)
+
+                elif type(val_59) is int:
+                    write.write_long(buffer, 1)
+                    write.write_int(buffer, val_59)
+
+    write.write_long(buffer, 0)
+    output.write(buffer)
```

Comment:
Isn't it possible to write into `output` right away?

Reply:
It is, however I ran benchmarks on the tests schemata in cerializer and it turns out that this approach is 22% faster than directly writing to output. But thanks, I did not think about that.